Build in warning dialogs for actions that change a users own groups

luisa-beerboom opened the pull request:
Closes #2688 

Built in warning dialogs for when a user
1. Throws himself out of a meeting using the account lists multiselect assign meetings feature
2. Removes some of his own groups via the participant detail view (and the admin group is not among the remaining ones)
2. Removes some of his own groups via the participant list info dialog (and the admin group is not among the remaining ones)

bastianjoel commented on client/src/app/site/pages/meetings/pages/participants/pages/participant-detail/components/participant-detail-view/participant-detail-view.component.ts:
```diff
@@ -323,7 +323,24 @@ export class ParticipantDetailViewComponent extends BaseMeetingComponent {
             };
-            await this.repo.update(payload, this.user!).concat(this.repo.setPresent(isPresent, this.user!)).resolve();
+            const title = this.translate.instant(`This action will remove you from one or more groups.`);
+            const content = `This may diminish your ability to do things in this meeting and you may not be able to revert it by youself. Are you sure you want to do this?`;
```
This should be translatable.
```suggestion
            const content = _(`This may diminish your ability to do things in this meeting and you may not be able to revert it by youself. Are you sure you want to do this?`);
```

bastianjoel commented on client/src/app/site/pages/meetings/pages/participants/pages/participant-detail/components/participant-detail-view/participant-detail-view.component.ts:
```diff
@@ -323,7 +323,24 @@ export class ParticipantDetailViewComponent extends BaseMeetingComponent {
                     : []
             };
-            await this.repo.update(payload, this.user!).concat(this.repo.setPresent(isPresent, this.user!)).resolve();
+            const title = this.translate.instant(`This action will remove you from one or more groups.`);
```
Translation seems to be done in prompt component. So using marker function should be enough here.
```suggestion
            const title = _(`This action will remove you from one or more groups.`);
```

bastianjoel commented on client/src/app/site/pages/meetings/pages/participants/pages/participant-list/components/participant-list/participant-list.component.ts:
```diff
@@ -186,15 +188,29 @@ export class ParticipantListComponent extends BaseMeetingListViewComponent<ViewU
                     result.vote_delegated_to_id = null;
                 }
-                this.repo.update(result, user).resolve();
+                const title = this.translate.instant(`This action will remove you from one or more groups.`);
+                const content = `This may diminish your ability to do things in this meeting and you may not be able to revert it by youself. Are you sure you want to do this?`;
+                if (
+                    !(
+                        user.id === this.operator.operatorId &&
+                        this.operator.user
+                            .group_ids()
+                            .filter(group => group !== this.activeMeeting.default_group_id)
+                            .some(id => !(result.group_ids ?? []).includes(id)) &&
+                        !result.group_ids.includes(this.activeMeeting.admin_group_id)
+                    ) ||
+                    (await this.prompt.open(title, content))
+                ) {
+                    this.repo.update(result, user).resolve();
+                }
```
This seems quite duplicated to the code in `participant-detail-view.component.ts`. Could this be done elsewhere? Maybe add an `isMeetingAdmin` method to the operator or active meeting service.

bastianjoel commented on client/src/app/site/pages/meetings/pages/participants/pages/participant-list/components/participant-list/participant-list.component.ts:
```diff
@@ -186,15 +188,29 @@ export class ParticipantListComponent extends BaseMeetingListViewComponent<ViewU
                 }
-                this.repo.update(result, user).resolve();
+                const title = this.translate.instant(`This action will remove you from one or more groups.`);
+                const content = `This may diminish your ability to do things in this meeting and you may not be able to revert it by youself. Are you sure you want to do this?`;
```
Translation marker missing.

Elblinator reviewed (changes requested):
After removing yourself as a meeting-admin while you are also an account-admin over the orga-level then you can still see the icons you can only see because you are a meeting admin. If you reload the icons disappear. 
They should disappear without reload.

You can still remove yourself (without warning) as meeting-admin in a meeting via participants > multi-select > add/remove groups

If you remove your admin status from the edit detail dialog (in the participant-list click on the group names of your participant), the dialog opens correctly, but afterwards you can still open this dialog, even if you lost the possibility to change it. And if you remove yourself as an admin then some participants (potentially nonsuper/orga/account-admins) have the icon for not-active accounts

you can still remove your admin status in your own participant-detail view.
if you remove your group without saving, then your own detail-view is missing the admin-group until reload 

You can still remove yourself as orga-admin and account admin with a warning.

luisa-beerboom commented:
> After removing yourself as a meeting-admin while you are also an account-admin over the orga-level then you can still see the icons you can only see because you are a meeting admin. If you reload the icons disappear. They should disappear without reload.

This is a separate bug

> You can still remove yourself (without warning) as meeting-admin in a meeting via participants > multi-select > add/remove groups

Fixed

> If you remove your admin status from the edit detail dialog (in the participant-list click on the group names of your participant), the dialog opens correctly, but afterwards you can still open this dialog, even if you lost the possibility to change it. And if you remove yourself as an admin then some participants (potentially nonsuper/orga/account-admins) have the icon for not-active accounts

Also a separate bug

> you can still remove your admin status in your own participant-detail view. if you remove your group without saving, then your own detail-view is missing the admin-group until reload

First part is fixed, second is a separate bug

> You can still remove yourself as orga-admin and account admin without a warning.

Fixed

Elblinator reviewed (approved):
function
